CASSGO-59 CASSGO-56 Test fix for hostpool package and newHostInfo() expose

tengu-alt opened the pull request:
hostpool package test was using recently removed `SetConnectAddress()` 
The test was refactored to create hostInfo by constructor func, and the `newHostInfo` constructor was exposed. 
Also fix for the CASSGO-56

joao-r-reis commented:
We should fix the CI issue here on this PR too so we know we're fixing the issue and there won't be regressions

tengu-alt commented:
> We should fix the CI issue here on this PR

Understood, I'll do it.

joao-r-reis commented:
@tengu-alt I don't see `TestHostPolicy_HostPool` in the CI log still, I think the fix is to add `./..` to `go test`, we should do it for all steps in the workflow I think

tengu-alt committed:
Test fix for hostpool package
hostpool package test was refactored to create HostInfo via exported NewhostInfo() constructor.
NewHostInfo() was exposed.

patch by Oleksandr Luzhniy; reviewed by João Reis, for CASSGO-59

tengu-alt commented:
> I think the fix is to add ./.. to go test, we should do it for all steps in the workflow I think

Agree, I've added the `./...` to jobs `go test`, also I've added the `unit` build tags for other packages tests(e.g. `murmur` and `lru`) to prevent double run during integration tests.

tengu-alt commented:
I had to remove the `./...` from integration test runs because we already have those within `env.args`

tengu-alt commented:
@joao-r-reis  It seems that `TestHostPolicy_HostPool` [is passed test](https://github.com/apache/cassandra-gocql-driver/actions/runs/13971227224/job/39392956606#step:5:456). Please take a look.
